tools/torch_quant/torch_quant/quantizer.py

```diff
@@ -0,0 +1,57 @@
+from enum import Enum
+from typing import Optional
+
+import torch.nn as nn
+from torch.fx import GraphModule, Tracer
+
+from torch_quant.graph import (fold_qdq, insert_act_observer, modify_graph,
+                               observer_to_qdq, q_ref_dq_to_fbgemm,
+                               quantizable_module_to_ref, set_qconfig)
+from torch_quant.module import ModuleFilter, copy_and_replace, fx_trace
+from torch_quant.observer import toggle_observer
+
+
+class Backend(Enum):
+    REFERENCE = 0
+    DISC = 1
+    FBGEMM = 2
+
+
+class Quantizer:
+    def __init__(self, module_filter: Optional[ModuleFilter] = None,
+                 backend: Optional[Backend] = Backend.REFERENCE,
+                 tracer: Optional[Tracer] = None) -> None:
+        self.module_filter = module_filter
+        self.backend = backend
+        self.tracer = tracer
+
+    def calib_gm(self,  gm: GraphModule, root: nn.Module) -> None:
+        modify_graph(gm, root, [set_qconfig, insert_act_observer])
+        toggle_observer(gm, observe=True, fake_quant=False)
+
+    def calib(self, model: nn.Module) -> nn.Module:
+        trace_mapping = fx_trace(model, self.module_filter, tracer=self.tracer)
+        for x in trace_mapping.values():
+            self.calib_gm(x.gm, x.m)
+        return copy_and_replace(model, trace_mapping)
+
+    def quantize_gm(self, gm: GraphModule, root: nn.Module) -> None:
+        passes = [set_qconfig, insert_act_observer,
+                  observer_to_qdq, quantizable_module_to_ref]
```

Comment:
Why replace observers with quantize/de-quantize? For blade-disc/blade-trt backends, replacing fake-quant with q-dq is done on torchscript.

Reply:
disc backend support has not been implemented yet in this initial commit. A simple implementation in my mind is: there will be different 'quantize passes' (or convert passes) for different backend, and common passes are those for Backend.REFERENCE. Of course it could be inappropriate or unnecessary to share common passes, I'll refactor this in next PR when supporting disc backend.